core/src/main/java/io/confluent/rest/ApplicationGroup.java

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.confluent.rest;
+
+import org.eclipse.jetty.server.Server;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class ApplicationGroup<T extends RestConfig> {
+  private final T config;
+  private Server server = null;
+  private final List<Application> applications = new ArrayList<>();
+
+  ApplicationGroup(T config, Application ...applications) {
+    this.config = config;
+    this.applications.addAll(Arrays.asList(applications));
+    System.out.println(applications.toString());
+  }
+
+  public void addApplication(Application application) {
+    this.applications.add(application);
+  }
+
+  public List<Application> getApplications() {
+    return applications;
+  }
+
+  public RestConfig getConfiguration() {
+    return this.config;
+  }
+
+  /* Visible for testing */
+  Server getServer() {
+    return this.server;
+  }
+
+  /**
+   * Start the server (creating it if necessary).
+   * @throws Exception If the application fails to start
+   */
+  public void start() throws Exception {
```

Comment:
Yeah it just felt weird to leave them off because the individual applications expose life cycle methods. You are right though we should remove it from the group so we can position ourselves to remove them from the group and deprecate the application lifecycle methods to move rest-utils closer to where we want it to be in the future. .